src/backend/joanie/tests/core/api/admin/orders/test_list.py

```diff
@@ -503,3 +506,308 @@ def test_api_admin_orders_list_filter_by_id(self):
+
+        response = self.client.get(
+            f"/api/v1.0/admin/orders/?created_on_after={isoformat_searched_date}"
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        content = response.json()
+        self.assertEqual(content["count"], 0)
+
+    def test_api_admin_order_list_filter_by_created_on_after(self):
+        """
+        Authenticated admin user should be able to get the list of orders that
+        were created after the given date.
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+
+        # Create orders with a date in the past
+        mocked_date = date(2024, 12, 19)
+        with mock.patch("django.utils.timezone.now", return_value=mocked_date):
+            orders = factories.OrderGeneratorFactory.create_batch(7)
+
+        isoformat_searched_date = date(2024, 12, 1).isoformat()
+
+        response = self.client.get(
+            f"/api/v1.0/admin/orders/?created_on_after={isoformat_searched_date}"
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        content = response.json()
+        response_ids = [order["id"] for order in content["results"]]
+        expected_ids = [str(order.id) for order in orders]
+        self.assertEqual(content["count"], 7)
+        self.assertListEqual(sorted(response_ids), sorted(expected_ids))
+
+    def test_api_admin_order_list_filter_by_created_on_before_and_after_date_gets_no_result(
+        self,
+    ):
+        """
+        Authenticated admin user should not be able to find orders because the
+        range of given date do not match any orders created.
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+
+        # Create orders with a date in the past
+        mocked_date_1 = date(2024, 11, 20)
+        with mock.patch("django.utils.timezone.now", return_value=mocked_date_1):
+            factories.OrderGeneratorFactory.create_batch(3)
+
+        # Create orders with a date in the past
+        mocked_date_2 = date(2024, 11, 30)
+        with mock.patch("django.utils.timezone.now", return_value=mocked_date_2):
+            factories.OrderGeneratorFactory.create_batch(2)
+
+        isoformat_after_date = date(2024, 12, 1).isoformat()
+        isoformat_before_date = date(2024, 12, 7).isoformat()
+
+        response = self.client.get(
+            "/api/v1.0/admin/orders/?"
+            f"created_on_date_range_after={isoformat_after_date}"
+            f"&created_on_date_range_before={isoformat_before_date}"
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        content = response.json()
+        self.assertEqual(content["count"], 0)
+
+    def test_api_admin_order_list_filter_by_created_on_before_and_after(self):  # pylint: disable=too-many-locals
```

Comment:
I feel like these two last tests are not necessary. WDYT?

Reply:
Those two last tests exposed a new way to use the filter by just passing one out of two awaited parameters for the date range search. I think I should keep those to show the possibilities.